### PR TITLE
#159395028 Setup Heroku Deployment

### PR DESCRIPTION
**Pull request by archbishop12**
#### What does this PR do?
- Sets up Heroku deployment 
#### Description of Task to be completed?
- Create Heroku application instance
- Provision Postgresql database instance
- Link instance to Github project branch
- Set environment variables and enable buildpacks to serve bower components in the project
- Deploy the application to Heroku
#### How should this be manually tested?
After cloning the repo, create a virtual environment and run the following commands
- pip install -r requirements_devel.txt
- npm install
- Set the following environment variables for Postgresql database
   - DB_NAME
   - DB_USER
   - DB_HOST
   - DB_PASSWORD

#### What are the relevant pivotal tracker stories?
[#159395028](https://www.pivotaltracker.com/story/show/159395028)

#### Screenshots
screen shot 2018-07-30 at 18 14 28
screen shot 2018-07-30 at 18 15 41

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -62,3 +61,7 @@ target/
 # External Libraries 
 wger/core/static/bower_components
 node_modules
+
+# IDE specific files
+.vscode/
+.DS_Store
```
[the remark this answers is not in the document]
Comment (Jpkat92) [reply .gitignore]: @isheebo this is done

File: Procfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+release: bash ./pre-release.sh
+web: gunicorn wger.wsgi:application
```
[the remark this answers is not in the document]
Comment (Jpkat92) [reply Procfile]: this is done

File: pre-release.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+echo "Running pre-release scripts"
+python setup.py develop
+npm install
+invoke bootstrap-wger --settings-path wger/settings.py --no-start-server
+python manage.py makemigrations --merge
+python manage.py migrate
+python manage.py bower_install --allow-root 
+echo "Done running pre-release.sh"
```
[the remark this answers is not in the document]
Comment (Jpkat92) [reply pre-release.sh]: this is done

**Comment by magicmarie** [comment]
@archbishop12 @Jpkat92 Fix travis builds. Pep8 is appearing twice in the different requirements files. Delete the one in requirements.txt and keep the one in requirements_devel. When requirements_devel runs, it always calls requirements.txt.

**Comment by mutebironald** [comment]
Fix Travis build @archbishop12 @Jpkat92

**Pull request closed without merging**